context/app/static/js/components/home-revision/AssayTypeBarChart/AssayTypeBarChart.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useContext, useMemo } from 'react';
+import { scaleLinear, scaleOrdinal, scaleBand } from '@visx/scale';
+import { BarStackHorizontal } from '@visx/shape';
+import { Group } from '@visx/group';
+import { AxisTop, AxisLeft } from '@visx/axis';
+import { withParentSize } from '@visx/responsive';
+
+import { AppContext } from 'js/components/Providers';
+import useSearchData from 'js/hooks/useSearchData';
+import { formatAssayData, addSumProperty, sortBySumAscending } from './utils';
+import { useChartPalette } from './hooks';
+
+const organTypesQuery = {
+  size: 0,
+  aggs: {
+    organ_types: { terms: { field: 'origin_sample.mapped_organ.keyword', size: 10000 } },
+  },
+};
+
+const assayTypesQuery = {
+  size: 0,
+  aggs: {
+    mapped_data_types: {
+      composite: {
+        sources: [
+          {
+            mapped_data_type: {
+              terms: {
+                field: 'mapped_data_types.keyword',
+              },
+            },
+          },
+          {
+            organ_type: {
+              terms: {
+                field: 'origin_sample.mapped_organ.keyword',
+              },
+            },
+          },
+        ],
+        size: 10000,
+      },
+    },
+  },
+};
+
+const getDataType = (d) => d.mapped_data_type;
+
+function AssayTypeBarChart({ parentWidth, parentHeight }) {
+  const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
+
+  const colors = useChartPalette();
+
+  const { searchData: organTypesData } = useSearchData(organTypesQuery, elasticsearchEndpoint, nexusToken);
+
+  const buckets = organTypesData?.aggregations?.organ_types?.buckets || [];
+  const organTypes = buckets.map((b) => b.key);
+
+  const { searchData: assayTypesData } = useSearchData(assayTypesQuery, elasticsearchEndpoint, nexusToken);
+
+  const { formattedData, maxSumDocCount } = useMemo(() => {
+    if (Object.keys(assayTypesData).length > 0) {
+      const f = addSumProperty(formatAssayData(assayTypesData));
```

Comment:
Strong preference for spelling out variables to at least some degree, at least in situations like this.

Reply:
Sounds good, I couldn't reuse `formattedData` or `maxSumDocCount` so I just opted for something quick. I'll think of different variable names.